tests/integration/cmd_devfile_init_test.go

```diff
@@ -377,30 +381,73 @@ var _ = Describe("odo devfile init command tests", func() {
 			})
 
 			When("devfile contains parent URI", func() {
-				var originalKeyList []string
-				var srcDevfile string
+				var childDevfile string
+				var originalDevfileContent string
 
 				BeforeEach(func() {
-					var err error
-					srcDevfile = helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-parent.yaml")
-					originalDevfileContent, err := os.ReadFile(srcDevfile)
+					// An issue from the Devfile library causes the parent to not be resolved properly:
+					//  ✗  unable to parse devfile: failed to populateAndParseDevfile: url path to directory or file should contain 'tree' or 'blob'.
+					//
+					// Even after using a URL with 'tree'or 'blob', the parser still does not pass.
+					// This workaround downloads the parent manually and references it via the local filesystem.
+					srcDir := helper.CreateNewContext()
+					DeferCleanup(func() {
+						helper.DeleteDir(srcDir)
+					})
+					parentDevfile := filepath.Join(commonVar.Context, "parent-devfile.yaml")
+					f, err := os.Create(parentDevfile)
+					Expect(err).ShouldNot(HaveOccurred())
+					resp, err := http.Get("https://github.com/OpenLiberty/application-stack/releases/download/maven-0.7.0/devfile.yaml")
+					Expect(err).ShouldNot(HaveOccurred())
+					defer resp.Body.Close()
+					_, err = io.Copy(f, resp.Body)
+					Expect(err).ShouldNot(HaveOccurred())
+					childDevfile = filepath.Join(srcDir, "devfile-child.yaml")
+					helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-parent.yaml"), childDevfile)
+					helper.ReplaceString(childDevfile, "https://github.com/OpenLiberty/application-stack/releases/download/maven-0.7.0/devfile.yaml", "parent-devfile.yaml")
+
+					b, err := os.ReadFile(childDevfile)
 					Expect(err).To(BeNil())
```

Comment:
Could you do this once and for all by modifying the `devfile-with-parent.yaml` directly in the odo repo, and copying in the odo repo the parent devfile?

Reply:
This change has been reverted, and this behavior should not change.